### PR TITLE
reach + frequency metrics

### PR DESCRIPTION
**Pull request by fivetran-jamie**
## PR Overview
**This PR will address the following Issue/Feature:**
https://github.com/fivetran/dbt_facebook_ads_source/issues/34

**This PR will result in the following new package version:**

v0.7.3

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Bug Fixes
- Resolved the "duplicate column" error that would arise when the `facebook_ads__basic_ad_passthrough_metrics` variable included `reach` or `frequency`. `Reach` and `frequency` are included by default in the `stg_facebook_ads__basic_ad` model but will only persist to downstream transform models if specified in the `facebook_ads__basic_ad_passthrough_metrics` variable.

## Under the Hood
- Updated the maintainer PR template to our most up-to-date standards.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

So in my dev `dbt_project.yml`, I added the following config:
```yml
vars:
  facebook_ads__basic_ad_passthrough_metrics: 
    - name: 'reach'
    - name: 'frequency'
      alias: 'fake_freq_name'
```

I expected the above inclusion of `reach` (but not `frequency` actually due to the alias) to cause a failure, which it did in `stg_facebook_ads__basic_ad`
image

With this branch, running `dbt run +ad_reporting` completed successfully 
image

And I see `reach` and `fake_freq_name` in my facebook ads models (and with non-zero/null values)
image
image

### If you had to summarize this PR in an emoji, which would it be?
 
📊

### CODE REVIEW

File: models/stg_facebook_ads__basic_ad.sql [hunk 1]
```diff
@@ -36,9 +36,29 @@ final as (
...
+                {% do check.append('frequency') %}
+            {% endif -%}
+        {% endfor -%}
+
+        {% if 'reach' not in check %}
+        , reach
+        {% endif %}
+        {% if 'frequency' not in check %}
+        , frequency
+        {%- endif %}
```
Comment (fivetran-joemarkiewicz) [inline models/stg_facebook_ads__basic_ad.sql:61]: It seems like we are doing a lot of hard conditional checks. I wonder if there is a way to make this more dynamic and less reliant on the hard coded conditional checks. 🤔 Similar to the approach you used in the macro.

Comment (fivetran-joemarkiewicz) [reply models/stg_facebook_ads__basic_ad.sql:61]: I just tested the following and it looks to achieve the same result without the hard conditional checks and less code. What are your thoughts?
```suggestion
        {%- set check = [] %}
        {%- for field in var('facebook_ads__basic_ad_passthrough_metrics') -%}
            {%- set field_name = field.alias|default(field.name)|lower %}
            {% if field_name in ['reach', 'frequency'] %}
                {% do check.append(field_name) %}
            {% endif %}
        {%- endfor %}

        {%- for metric in ['reach', 'frequency'] -%}
            {% if metric not in check %}
                , {{ metric }}
            {% endif %}
        {%- endfor %}
```

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-jamie thanks for quickly working through this update and for putting such a strong emphasis on backwards compatibility! I do have one small comment to try and make the code a bit more dynamic in the staging model. Let me know your thoughts.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-jamie thanks for making these changes! This looks good to me! This is good to have docs regen'd and then kick off the release review!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,12 @@
+# dbt_facebook_ads_source v0.7.3
+[PR #35](https://github.com/fivetran/dbt_facebook_ads_source/pull/35) includes the following updates:
+
+## Bug Fixes
+- Resolved the "duplicate column" error that would arise when the `facebook_ads__basic_ad_passthrough_metrics` variable included `reach` or `frequency`. `Reach` and `frequency` are included by default in the `stg_facebook_ads__basic_ad` model but will only persist to downstream transform models if specified in the `facebook_ads__basic_ad_passthrough_metrics` variable.
```
Comment (fivetran-avinash) [inline CHANGELOG.md:5]: Should we call out how we resolved it in a sub-bullet using a for loop in the specific models/macros? Doesn't have to be too involved, just always nice to explain how to the customer.

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-jamie A few comments before approval!

File: CHANGELOG.md (hunk 2 above)
Comment (fivetran-jamie) [reply CHANGELOG.md]: added!

File: CHANGELOG.md [hunk 3]
```diff
@@ -23,10 +32,10 @@
 
 # dbt_facebook_ads_source v0.UPDATE.UPDATE
 
- ## Under the Hood:
-
+## Under the Hood:
 - Incorporated the new `fivetran_utils.drop_schemas_automation` macro into the end of each Buildkite integration test job.
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply CHANGELOG.md:33]: updated!

**Review by fivetran-avinash — APPROVED** [review]
lgtm

**Pull request merged**